api/types.proto

```diff
@@ -447,13 +447,25 @@ message AcceptancePolicy {
 }
 
 
+message ExternalCA {
+	enum CAType {
+		CFSSL = 0 [(gogoproto.enumvalue_customname) = "CATypeCFSSL"];
+	}
+
+	// CAType is the type of this external CA.
+	CAType type = 1;
+
+	// URL is the URL where the external CA can be reached.
+	string url = 2 [(gogoproto.customname) = "URL"];
```

Comment:
See https://github.com/docker/swarmkit/pull/1099#issuecomment-229518249 for more questions/thinking on this matter but I think it's essential that external CA options are supported from the beginning as any external CA will likely require them. This includes CFSSL for what is likely the normal case of not wanting root CA information to be shared between it and Swarm.